### PR TITLE
refactor(blocks): Migrate `blocks/loops.js` to TypeScript

### PR DESCRIPTION
**Pull request by rachel-fenichel**
## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)
- [x] I ran `npm run format` and `npm run lint`

### Resolves

Part of #6828 

### Proposed Changes

Migrates loop blocks, following the examples in #6900 and #6901

### CODE REVIEW

File: blocks/loops.ts [hunk 1]
```diff
@@ -303,20 +305,24 @@ Extensions.register(
...
+/** Type of a block that has CONTROL_FLOW_IN_LOOP_CHECK_MIXIN */
+type ControlFlowInLoopBlock = Block&ControlFlowInLoopMixin;
+interface ControlFlowInLoopMixin extends ControlFlowInLoopMixinType {};
+type ControlFlowInLoopMixinType = typeof CONTROL_FLOW_IN_LOOP_CHECK_MIXIN;
+
 /**
  * This mixin adds a check to make sure the 'controls_flow_statements' block
  * is contained in a loop. Otherwise a warning is added to the block.
+ *
  * @mixin
```
Comment (cpcallen) [inline blocks/loops.ts:326]: See above.

File: blocks/loops.ts [hunk 2]
```diff
@@ -342,18 +348,16 @@ const CONTROL_FLOW_IN_LOOP_CHECK_MIXIN = {
...
     // Don't change state if:
     //   * It's at the start of a drag.
     //   * It's not a move event.
-    if (!this.workspace.isDragging || this.workspace.isDragging() ||
-        e.type !== Events.BLOCK_MOVE) {
+    if (!ws.isDragging || ws.isDragging() || e.type !== Events.BLOCK_MOVE) {
       return;
     }
-    const enabled = this.getSurroundLoop(this);
+    const enabled = !!this.getSurroundLoop();
```
Comment (cpcallen) [inline blocks/loops.ts:360]: Good for clarity!  I'm curious: was this necessary to get the ternary below to type-check?

**Review by cpcallen — CHANGES_REQUESTED** [review]
LGTM except for a few @directives that should be removed.  A few additional comments with queries for my curiosity.

File: blocks/loops.ts [hunk 3]
```diff
@@ -6,42 +6,32 @@
...
  * @fileoverview Loop blocks for Blockly.
- * @suppress {checkTypes}
  */
 
 import * as goog from '../closure/goog/goog.js';
 goog.declareModuleId('Blockly.libraryBlocks.loops');
 
-/* eslint-disable-next-line no-unused-vars */
-import type * as AbstractEvent from '../core/events/events_abstract.js';
+import type {Abstract} from '../core/events/events_abstract.js';
```
[the remark this answers is not in the document]
Comment (rachel-fenichel) [reply blocks/loops.ts]: Done

**Pull request merged**